### PR TITLE
feat(jax): export call_lower to SavedModel via jax2tf

### PR DESCRIPTION
**Pull request by njzjz**
## Release Notes

- **New Features**
	- Added support for the TensorFlow SavedModel format, allowing users to handle additional model file types.
	- Introduced a new TensorFlow model wrapper class for enhanced integration with JAX functionalities.
  
- **Bug Fixes**
	- Improved error handling for unsupported file formats during model deserialization.

- **Documentation**
	- Updated backend documentation to reflect new file extensions and clarify backend capabilities.

- **Tests**
	- Enhanced test structure for better clarity and maintainability regarding backend handling.
	- Added a new job for testing TensorFlow 2 in eager mode within the testing workflow.
	- Introduced a conditional skip for tests based on TensorFlow 2 compatibility.

### CODE REVIEW

File: deepmd/jax/jax2tf/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import tensorflow as tf
+
+if not tf.executing_eagerly():
+    # TF disallow temporary eager execution
+    raise RuntimeError(
+        "Unfortunatly, jax2tf (requires eager execution) cannot be used with the "
+        "TensorFlow backend (disables eager execution). "
+        "If you are converting a model between different backends, "
+        "considering converting to the `.dp` format first."
+    )
```
Comment (njzjz) [inline deepmd/jax/jax2tf/__init__.py:11]: Due to this reason, I cannot enable the test in the CI. The test passed locally when I temporarily disabled the TF.

File: deepmd/jax/jax2tf/tfmodel.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        """Get the model definition script."""
+        return self.model_def_script
+
+    def get_min_nbor_dist(self) -> Optional[float]:
+        """Get the minimum distance between two atoms."""
+        return self.min_nbor_dist
+
+    def get_nnei(self) -> int:
+        """Returns the total number of selected neighboring atoms in the cut-off radius."""
+        return self.nsel
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/jax2tf/tfmodel.py:269]:
```suggestion
    def get_nnei(self) -> int:
        """Returns the total number of selected neighboring atoms in the cut-off radius."""
        return self.get_nsel()
```

**Comment by njzjz** [comment]
I found XLA ([deps here](https://github.com/jax-ml/jax/blob/ec39b592f7c096b0b8183723feaab2ed0d001041/examples/jax_cpp/BUILD#L24-L31)) is too hard to build, so I still prefer JAX2TF for C++ deployment.

**Review by wanghan-iapcm — COMMENTED** [review]
Before we have supported the savedmodel in deepEval, shall we write a specific test for it under tf2?

**Comment by njzjz** [comment]
> Before we have supported the savedmodel in deepEval, shall we write a specific test for it under tf2?

DeepEval has been supported in this PR. The test is added, but due to incompatibility between TF1 and TF2, they cannot be run in the same session.

File: source/tests/utils.py [hunk 3]
```diff
@@ -8,3 +8,4 @@
 
 # see https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables
 CI = os.environ.get("CI") == "true"
+DP_TEST_TF2_ONLY = os.environ.get("DP_TEST_TF2_ONLY") == "1"
```
Comment (wanghan-iapcm) [inline source/tests/utils.py:11]: should the default be false?

Comment (njzjz) [reply source/tests/utils.py:11]: Default is false.

**Pull request merged**